### ISSUE TITLE
IDE0300 incorrectly suggested for an attribute argument initializer

### ISSUE THREAD

**Issue by f2bo** [dotnet/roslyn#70058]
**Version Used**: 
8.0.0-rc.1.23419.4

**Steps to Reproduce**:
The following code suggests changing the array initialization of the `CallConvs` argument with a collection expression (`IDE0300: Collection initialization can be simplified`).

``` C#
[LibraryImport("mylib")]
[UnmanagedCallConv(CallConvs = new Type[] { typeof(System.Runtime.CompilerServices.CallConvCdecl) })]
public static partial void Foo();
``` 

**Diagnostic Id**:
`IDE0300`

**Expected Behavior**:
No suggestion

**Actual Behavior**:
Applying this fix results in the following code, which is not valid for an attribute argument (`Error CS0182 - An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type`).

``` C#
[UnmanagedCallConv(CallConvs = [typeof(System.Runtime.CompilerServices.CallConvCdecl)])]
```

**Comment by CyrusNajmabadi** [comment]
@jcouv  @cston  i believe you have a compiler fix for this.

Note: teh analyzer/fixer is correct.  As per the language specification, this is legal.  However, there is a compiler bug (being fixed now) which is flagging this as not allowed.

**Comment by f2bo** [comment]
Good to hear that this will be allowed. Thanks!

**Comment by mavasani** [comment]
Moving to compiler team and assigning to Julien/Chuck as per Cyrus' comment.

**Comment by jcouv** [comment]
> However, there is a compiler bug (being fixed now) which is flagging this as not allowed.

@CyrusNajmabadi Would you have a link by any chance?

**Comment by CyrusNajmabadi** [comment]
@jcouv ... didn't you fix this?

**Comment by jcouv** [comment]
Oh, I get it (thanks @cston for the offline clarification).
Yes, collection expressions are supported in attributes as of PR https://github.com/dotnet/roslyn/pull/69968
I'll go ahead and close this as duplicate of https://github.com/dotnet/roslyn/issues/69133
